### PR TITLE
CI: Automate release flow.

### PR DESCRIPTION
**Pull request by ami-descope**
## Description

Automate release process to bump version and repoint the tag to the latest commit

### CODE REVIEW

File: .github/workflows/publish-gem.yaml [hunk 1]
```diff
@@ -24,6 +24,28 @@ jobs:
...
+          git config --global user.email 'github-actions@github.com'
+          git commit -am "Bump version to $NEW_VERSION"
+          git push
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Repoint the tag to latest commit
+        run: |
+          git tag -fa ${{ github.event.release.tag_name }} -m "Release $NEW_VERSION"
+          git push origin :${{ github.event.release.tag_name }}
```
Comment (tomski747) [inline .github/workflows/publish-gem.yaml:44]: Tag is not a branch, you cannot push commits to a tag. 
As far as i understand, to release a new version you need to :
1. Create a "bump version file" PR
2. Bump
3. Approve and merge
4. Tag main with new version
5. Github Release

Comment (ami-descope) [reply .github/workflows/publish-gem.yaml:44]: the flow is:

Yael is creating a new tag and release

the release is bumping the version.rb with that tag

now commit that change

then repoint the tag to that commit

Comment (ami-descope) [reply .github/workflows/publish-gem.yaml:44]: `git push origin :${{ github.event.release.tag_name }}` is DELETING the tag so it can retag

Comment (ami-descope) [reply .github/workflows/publish-gem.yaml:44]: so this code

```yaml
git tag -fa ${{ github.event.release.tag_name }} -m "Release $NEW_VERSION"
git push origin :${{ github.event.release.tag_name }}
git push
```

is force create tag on the new commit from that bump
then delete the tag from remote
then push the changes which will push the tag that is now pointing to the code with the new version.rb
all of this happens only on release flow

**Review by tomski747 — APPROVED** [review]

**Pull request merged**